src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { getUserById } from '../../services/user';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  setNewTodos: (val: any) => void;
+  newTodos: Todo[];
+};
+
+export const TodoForm: React.FC<Props> = ({ setNewTodos, newTodos }) => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const addTodo = (newPreparedTodo: Todo) => {
+      setNewTodos((currentTodos: any) => [
+        ...currentTodos,
+        { ...newPreparedTodo },
+      ]);
+    };
+
+    const newPreparedTodo: Todo = {
+      id: Math.max(...newTodos.map(t => t.id)) + 1,
```

Comment:
Would be better to move it to separate function instead of writing it in the object directly